data/default.nix

```diff
@@ -6,27 +6,47 @@ with (import nixpkgs config);
 
 let
   mkDerivation =
-    { srcs ? ./elm-srcs.nix
-    , src
-    , name
+    { elmSrcs ? ./elm-srcs.nix
+    , srcs
     , srcdir ? "./src"
+    , extension ? ".elm"
```

Comment:
`elm make` doesn't care about the extension of entry file. For instance, some people like to name the main file as `Main.js.elm` - This doesn't break `elm make Main.js.elm` but it prevents the module from being imported into another module with `import Main`. I think it's a good idea to make this easily overridable.